clientv3/watch.go

```diff
@@ -46,6 +46,20 @@ type Watcher interface {
 	// through the returned channel. If revisions waiting to be sent over the
 	// watch are compacted, then the watch will be canceled by the server, the
 	// client will post a compacted error watch response, and the channel will close.
+	// If the context "ctx" is canceled or timed out, returned "WatchChan" is closed,
+	// and "WatchResponse" from this closed channel has zero events and nil "Err()".
+	// If the context is "context.Background/TODO", returned "WatchChan" will not be closed
+	// and wait until events happen, except when server returns an error and:
```

Comment:
except when server returns non-recoverable error.

When require leader is passed in, watch will return `no leader` error when the connected server has no leader.

you probably need to polish my words, but you get the idea.